osu.Game/Overlays/Settings/SettingsItem.cs

```diff
@@ -2,26 +2,52 @@
 // Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
 
 using System.Collections.Generic;
+using osu.Framework.Allocation;
 using OpenTK.Graphics;
 using osu.Framework.Configuration;
+using osu.Framework.Extensions.Color4Extensions;
 using osu.Framework.Graphics;
+using osu.Framework.Graphics.Colour;
 using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
 using osu.Framework.Graphics.Sprites;
 using osu.Framework.Graphics.UserInterface;
+using osu.Framework.Input;
+using osu.Game.Graphics;
 using osu.Game.Graphics.Sprites;
 
 namespace osu.Game.Overlays.Settings
 {
-    public abstract class SettingsItem<T> : FillFlowContainer, IFilterable
+    public abstract class SettingsItem<T> : Container, IFilterable
     {
         protected abstract Drawable CreateControl();
 
         protected Drawable Control { get; }
 
         private IHasCurrentValue<T> controlWithCurrent => Control as IHasCurrentValue<T>;
 
+        protected override Container<Drawable> Content => FlowContent;
+
+        protected readonly FillFlowContainer FlowContent;
+
         private SpriteText text;
 
+        private readonly SettingsItemDefaultIndicator<T> defaultIndicator = new SettingsItemDefaultIndicator<T>();

```

Comment:
Since this actually works as a button (and has logic built-in) I'd probably call it `RestoreDefaultValueButton` or similar